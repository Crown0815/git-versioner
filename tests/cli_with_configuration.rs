mod common;

use common::{cli, repo, TestRepo};
use indoc::formatdoc;
use insta::with_settings;
use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::fs;
use std::process::Command;

const CUSTOM_MAIN_BRANCH: &str = "stem";

#[macro_export]
macro_rules! assert_configured_repo_cmd_snapshot {
    ($repo:expr, $config:expr, $cmd:expr) => {
        with_settings!(
            { description =>
                formatdoc!(
                    "
                    Git Graph:
                        {graph}
                    Configuration ({path}):
                        {config}",
                    graph = $repo.graph(),
                    path = $config.file_name().unwrap().to_string_lossy(),
                    config = fs::read_to_string(&$config).unwrap()
                )
            },
            { assert_cmd_snapshot!($cmd); }
        );
    };
}

#[rstest]
fn test_release_candidate_on_toml_configured_main_branch(
    #[with(CUSTOM_MAIN_BRANCH)] mut repo: TestRepo,
    mut cli: Command
) {
    repo.cli_config.main_branch = Some(format!("^{}$", CUSTOM_MAIN_BRANCH));
    let config_file = repo.create_default_toml_config();

    assert_configured_repo_cmd_snapshot!(repo, config_file, cli.current_dir(&repo.path));
}
