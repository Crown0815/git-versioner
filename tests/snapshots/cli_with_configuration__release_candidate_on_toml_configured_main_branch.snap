---
source: tests/cli_with_configuration.rs
description: "Git Graph:\n    * 9d651ec (HEAD -> stem) 0.1.0-rc.1\n\nConfiguration (.git-versioner.toml):\n    main_branch = \"^stem$\"\n"
info:
  program: git-versioner
  args: []
---
success: true
exit_code: 0
----- stdout -----
{
  "Major": 0,
  "Minor": 1,
  "Patch": 0,
  "MajorMinorPatch": "0.1.0",
  "PreReleaseTag": "rc.1",
  "PreReleaseTagWithDash": "-rc.1",
  "PreReleaseLabel": "rc",
  "PreReleaseLabelWithDash": "-rc",
  "PreReleaseNumber": "1",
  "BuildMetadata": "",
  "SemVer": "0.1.0-rc.1",
  "AssemblySemVer": "0.1.0",
  "FullSemVer": "0.1.0-rc.1",
  "InformationalVersion": "0.1.0-rc.1"
}

----- stderr -----
